crates/operator-macros/src/lib.rs

```diff
@@ -0,0 +1,47 @@
+/// Creates a label from the provided string literal. Kubernetes labels
+/// can contain any valid ASCII data. It additionally must follow Kubernetes
+/// specific rules documented [here][k8s-labels].
+///
+/// [k8s-labels]: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/
+///
+/// ```
+/// let label = label!("stackable.tech/vendor=Stackable");
+/// ```
+#[macro_export]
+macro_rules! label {
+    ($Input:literal) => {{
+        use std::str::FromStr;
+        stackable_operator::kvp::Label::from_str($Input)
+    }};
+}
+
+/// Creates an annotation from the provided string literal. Kubernetes
+/// annotations can contain any valid UTF-8 data.
+///
+/// ```
+/// let annotation = annotation!("stackable.tech/vendor=Hello Wörld!");
+/// ```
+#[macro_export]
+macro_rules! annotation {
+    ($Input:literal) => {{
+        use std::str::FromStr;
+        stackable_operator::kvp::Annotation::from_str($Input)
+    }};
+}
```

Comment:
Why are these macros at all, actually?

Reply:
A shorter way of constructing a label / annotation. The idea was that this can eventually be used in const environments like this:

```rust
pub const APP_NAME = label!("app.kubernetes.io/name=airflow");
```

Reply:
That just depends on it being `const`. The value that we can provide would be compile-time validation... but the current API is incompatible with that anyway since it returns a `Result<Value>` rather than the `Value` itself.

Reply:
Yes that depends on it being `const`, which is a whole other discussion. When we would eventually have `const` support, we could use `panic` or compile-time result evaluation which would fail compilation on `Err(_)`.

Of course we can remove these macros for this PR. I just found them neat :)

Reply:
Compile-time validation would be the use-case I can see for macroing it.. but even then I'm honestly not sure how common it'd be to have a fully static label set?

Reply:
Yeah this is mostly useful for stuff like `stackable.tech/vendor=Stackable` or `stackable.tech/managed-by=stackablectl`.

Reply:
Yup. My question would be how often we set those without also setting the corresponding stacklet annotations.

Reply:
The macros as defined in this PR don't help anyone and might as well be removed. I'm not entirely against adding compile-time validated macros in the future, as long as we can minimize the maintenance burden that they add (as in: we should make sure that we can reuse the same validation infra on both sides).

Reply:
Okay, then I will remove the macros for this PR. Supporting them in the future shouldn't need much maintenance work.

Reply:
Removed in 7e6279d